docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,133 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1 (intentionally "naked" until we agree on the high-level design)
+
+## 1. Introduction
+Goal is to describe how we can extend the InstructLab CLI to include options to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output.
+
+References:<br/>
+**They will be removed later as they can't stay in a public repo**
+* [RAG Artifacts with RHEL AI](https://docs.google.com/document/d/1nALKAawvvNKlTnbeQIXmpIXbo2Lxm2vIw1euxW96TgE)
+* [Productization RHEL AI RAG Atifacts](https://docs.google.com/document/d/13lU1bKlqmihEyZDcYCnv4fiv4WUKFELNRXucDgm1d9M)
+
+## 2. Use Case
+TODO
+
+## 3. Proposed Commands
+### 3.1 RAG Ingestion Pipeline Command
+The proposal is to add a `rag` subgroup under the `data` group, with an `ingest` command, like:
+```
+ilab data rag ingest DOCS
+```
+
+#### Command Purpose
+This command processes embeddings generated from documents located in the *DOCS* folder and stores them in a vector database. These embeddings are intended to be used as augmented context in a Retrieval-Augmented Generation (RAG) chat pipeline.
+
+#### Assumptions
+The documents must be in JSON format and pre-processed using the docling tool.
+**TODO** what exact schema we expect and to document how to generate these formatted documents
+
+#### Supported Databases
+The command supports multiple vector database types. By default, it uses a local `MilvusLite` instance stored at `./rag-output.db`.
+
+#### Usage
+The generated embeddings can later be retrieved to enrich the context for RAG-based chat pipelines.
+
+### 3.2 RAG Ingestion Pipeline Options
```

Comment:
as mentioned above, we should first suggest applying the below options in the config file